### ISSUE TITLE
Upgrade to PhotoSwipe 5.x for additional a11y improvements

### ISSUE THREAD

**Issue by dnkemper** [uiowa/uiowa#4777]
waiting for TM's approval. Creating as followup issue to https://github.com/uiowa/uiowa/pull/4708

Capturing some additional improvements:

- font-size change for mobile captions
- consider https://github.com/dimsemenov/photoswipe-dynamic-caption-plugin
```[tasklist]
### Tasks
- [x] Image gallery block
- [x] Article image gallery
- [x] v2: Image gallery paragraph (config + SCSS)
- [x] Housing: residence halls (multiple fields) + floor plan paragraph type
- [x] IISC: article gallery
- [x] 91st Meridian: article gallery
- [x] ITS: article gallery
- [x] Iowa Now: article gallery
- [x] Articles narrow split: article gallery
- [x] Periodical split: article gallery
- [x] Classrooms: Room content type
- [ ] Remove image gallery view mode
```

**Comment by joewhitsitt** [comment]
noticed that exiting the gallery switches the tab back to the top of the page. the user would need to re-tab down and/or skip to main content to get back to the gallery. this is different than what we had for colorbox.

photoswipe is working on a v5 that supposedly has some other a11y improvements but v4 is currently what ships with the drupal module. https://www.drupal.org/project/photoswipe/issues/3232070

**Comment by dnkemper** [comment]
you're a wizard, Harry

**Comment by joewhitsitt** [comment]
There is now a RC for 5.x and we can evaluate what improvements have been made and see what else can be done.

**Comment by joewhitsitt** [comment]
There is now a stable release for this. https://github.com/uiowa/uiowa/pull/7531

Could also check to see if we can get lazy support back https://github.com/uiowa/uiowa/issues/6965

**Comment by joewhitsitt** [comment]
@pyrello heard you mention the grouping patch in standup. see this comment: https://github.com/uiowa/uiowa/issues/5332#issuecomment-1203180234

have you tested that the update breaks functionality?

**Comment by pyrello** [comment]
> @pyrello heard you mention the grouping patch in standup. see this comment: [#5332 (comment)](https://github.com/uiowa/uiowa/issues/5332#issuecomment-1203180234)
> 
> have you tested that the update breaks functionality?

Not sure if I said "patch" but I didn't mean to as I had already checked to make sure we weren't using any patches. I checked and this breaks the grouping functionality. It appears that there is some support for grouping in the 5.x branch: https://www.drupal.org/project/photoswipe/issues/3401194. Not sure if it is only loaded for views, but it is not working with our implementation, which seems to line up with what is suggested in that issue.

**Comment by joewhitsitt** [comment]
Looks like maybe we need to just reconfigure our implementation.

\Drupal\photoswipe\Plugin\Field\FieldFormatter\PhotoswipeFieldFormatter::view is adding the class to each item because the formatter is selected for the image gallery viewmode display settings. This should probably switch to a responsive image style and the parent field should be switched to use photoswipe responsive. This would confirm https://github.com/uiowa/uiowa/issues/4891 and we could finish that as part of this.

<img width="1032" alt="Screenshot 2024-05-30 at 11 22 56 AM" src="https://github.com/uiowa/uiowa/assets/4663676/4e12b643-886b-489c-9445-a9794179da2f">

I was just focusing on the v3 block implementation. we have other implementations (v2, housing, etc)

**Comment by pyrello** [comment]
I'm running into some issues with the v2 implementation that I haven't been able to figure out yet. Screenshots below using @bspeare's test page https://hawkeyemarchingband.stage.drupal.uiowa.edu/test-page:

Stage:
![Screenshot 2024-05-30 at 3 00 54 PM](https://github.com/uiowa/uiowa/assets/709009/58469012-37aa-47e8-ad33-6b2b54ba04ad)

Local:
![Screenshot 2024-05-30 at 3 00 36 PM](https://github.com/uiowa/uiowa/assets/709009/97a8afa5-f23d-4382-a083-bd109b037386)

This is probably something obvious, I just haven't figured it out yet.

**Comment by joewhitsitt** [comment]
Is it the switch from medium_nocrop to medium_widescreen?

**Comment by bspeare** [comment]
I just tested this on housing and the images are not clickable.  I also tested this on classrooms and got this error: `TypeError: Drupal\photoswipe\PhotoswipeAssetsManager::__construct(): Argument #2 ($module_handler) must be of type Drupal\Core\Extension\ModuleHandlerInterface, Drupal\Core\Render\Renderer given, called in /var/www/html/docroot/core/lib/Drupal/Component/DependencyInjection/Container.php on line 259 in Drupal\photoswipe\PhotoswipeAssetsManager->__construct() (line 59 of modules/contrib/photoswipe/src/PhotoswipeAssetsManager.php).`

**Comment by pyrello** [comment]
> I just tested this on housing and the images are not clickable. I also tested this on classrooms and got this error: `TypeError: Drupal\photoswipe\PhotoswipeAssetsManager::__construct(): Argument #2 ($module_handler) must be of type Drupal\Core\Extension\ModuleHandlerInterface, Drupal\Core\Render\Renderer given, called in /var/www/html/docroot/core/lib/Drupal/Component/DependencyInjection/Container.php on line 259 in Drupal\photoswipe\PhotoswipeAssetsManager->__construct() (line 59 of modules/contrib/photoswipe/src/PhotoswipeAssetsManager.php).`

I have not made any adjustments to housing yet.

**Comment by pyrello** [comment]
I said I thought I was almost done with this, but actually, I think there is a bit more to do with it.

**Comment by pyrello** [comment]
Since it is seeming less likely that I will get this done before I am gone for the weekend and in case someone else picks this up while I'm out: To figure out where there are implementations that need to be updated, I've been using PhpStorm's "Find in  Files" feature and searching for `view_mode: image_gallery`.

The switch that needs to happen is that, instead of using the Image Gallery view mode, which previously (and conveniently) contained all the photoswipe settings, we now need to use the "PhotoSwipe Responsive" formatter directly in all those field instances. I think after all these have been swapped, we can just get rid of the "Image Gallery" media view mode and its displays.

**Comment by pyrello** [comment]
I've not had luck with figuring out getting the captions to work yet, as detailed here: https://github.com/uiowa/uiowa/pull/7531#issuecomment-2148366532

I'm un-assigning myself for now.

**Comment by pyrello** [comment]
There has been some movement or at least a response to my question about captions: https://www.drupal.org/project/photoswipe/issues/3453071#comment-15648481

**Comment by pyrello** [comment]
As best as I can tell from looking into this again + re-reading the comments here, it seems that we are in a spot where we can get captions OR grouping to work without custom code. In the current iteration of working on this (#8816) I was able to select the correct field to populate the caption and that just works. However, I did not go through the process outlined in https://github.com/uiowa/uiowa/issues/4777#issuecomment-2140160924 to get the grouping working and I suspect that doing that will cause the captions to fail. Making a note of this so that I can remember when I pick this up again next week. It took me a little while to find the thread above to get the necessary context to recall what had to be done previously.

**Comment by pyrello** [comment]
After spending some more time looking into this, I think it should be prioritized as its own work. It would potentially be worth doing some more fiddling around with config for fields at various levels to see if there is a way to avoid custom code. But it seems like whatever solution we use is going to require custom code.

**Comment by briand44** [comment]
## Grooming
 
**Quadrant:** 

complicated/complex - some unknowns may have been figured out previously but lots to test

 **Discussion:**
 - should we tackle higher value updates (photoswipe?)
 - give us idea of what a high update will take
 - close on getting this updated last time
 - Notes from epic: “library version bump, open issues ([#4777](https://github.com/uiowa/uiowa/issues/4777), [#6965](https://github.com/uiowa/uiowa/issues/6965)) and patches” on why high
 - prior issues with grouping functionality and captions working together
 - they did away with our ability to use the title attribute
   - may have introduced a token as a way of solving for it

**Proposed solution or definition of done:** 
- update photoswipe to latest version
- make sure grouping functionality and captions are able to work together
- make sure we are not introducing regressions
- see if this update resolves issue that Marie is going to comment about
